lib/analyst/src/analyst.c

```diff
@@ -1,51 +1,57 @@
 #include "analyst/include/analyst.h"
 #include <assert.h>
+#include <fcntl.h>
 #include <sched.h>
 #include <signal.h>
 #include <stdint.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <unistd.h>
 
-Analyst create_analyst(char *lng_file_path, uint8_t max_number_of_int_to_read,
-                       FILE *output_stream) {
+Analyst create_analyst(char *lng_file_path, uint8_t max_number_of_int_to_read) {
   assert(lng_file_path != NULL && "empty lng file path");
 
+  sem_t *sem_block = sem_open("/sem_block", O_RDONLY);
+  assert(sem_block != SEM_FAILED && "failed to create sem_block semaphore");
+
+  FILE *lng_file = fopen(lng_file_path, "r");
+  assert(lng_file != NULL && "failed to open lng file");
+
   Analyst analyst = {
-      .current_line_number = 0,
-      .lng_file_path = lng_file_path,
       .max_number_of_int_to_read = max_number_of_int_to_read,
-      .output_stream = output_stream,
+      .sem_block = sem_block,
+      .lng_file = lng_file,
   };
 
   return analyst;
 }
 
 void start_analyst(Analyst *self) {
-  FILE *lng_file = fopen(self->lng_file_path, "rb");
-  assert(lng_file != NULL && "failed to open lng file");
 
   raise(SIGSTOP);
 
   pid_t buffer[self->max_number_of_int_to_read];
   unsigned long remaining_int;
 
   do {
+    sem_wait(self->sem_block);
+
     remaining_int =
-        fread(buffer, sizeof(pid_t), self->max_number_of_int_to_read, lng_file);
+        fread(buffer, sizeof(pid_t), self->max_number_of_int_to_read, self->lng_file);
```

Comment:
in the last iteration the analyst could read < `max_number_of_int_to_read` PIDs if `"total number of clients attended" % max_number_of_int_to_read == 0`. Instead, we should read the entire file since the max number of clients to read is guaranteed by the attendant.

Reply:
num intindi